lib/proxy/plugin_validators.rb

```diff
@@ -42,4 +42,15 @@ def validate!(settings)
       true
     end
   end
+
+  class PresenceAllowEmpty < Base
+    def validate!(settings)
+      setting_value = settings[@setting_name]
+
+      empty_value = setting_value.nil?
+
+      raise ::Proxy::Error::ConfigurationError, "Parameter '#{@setting_name}' is expected to be present" if empty_value
```

Comment:
I think the name `empty_value` here is misleading because the class name states it allows empty but the test fails if it's empty.